### PR TITLE
Add a better exception message

### PR DESCRIPTION
**Pull request by mwadams**
Adds to the existing error message to better explain the potential problem.
Adds specs to validate the common cases.
Derives from a more sensible base exception.

### CODE REVIEW

File: Solutions/Menes.Abstractions/Menes/Exceptions/OutputBuilderNotFoundException.cs [hunk 1]
```diff
@@ -4,78 +4,137 @@
...
-            sb.Append(". Recieved a Response with Status Code of ");
+            sb.Append(".");
 
-            if (this.RawResult is OpenApiResult oaiResult)
+            if (result is OpenApiResult oaiResult)
             {
-                this.OpenApiResult = oaiResult;
-
+                sb.AppendLine();
+                sb.Append("Recieved a Response with a Status Code of: ");
```
Comment (jongeorge1) [inline Solutions/Menes.Abstractions/Menes/Exceptions/OutputBuilderNotFoundException.cs:71]: Received

File: Solutions/Menes.Abstractions/Menes/Exceptions/OutputBuilderNotFoundException.cs [hunk 2]
```diff
@@ -4,78 +4,137 @@
...
+                        appendComma = true;
+                    }
+
+                    sb.AppendLine();
+                }
+            }
+            else
+            {
+                sb.AppendLine();
+                sb.Append("Recieved a Response with a type of ");
```
Comment (jongeorge1) [inline Solutions/Menes.Abstractions/Menes/Exceptions/OutputBuilderNotFoundException.cs:131]: Received

Comment (mwadams) [reply Solutions/Menes.Abstractions/Menes/Exceptions/OutputBuilderNotFoundException.cs:131]: This strongly leads me to suspect I didn't type this code in the first place. It all looked really weird and I don't remember writing it.

Maybe I was possessed?

File: Solutions/Menes.Abstractions/Menes/Exceptions/OutputBuilderNotFoundException.cs (hunk 1 above)
Comment (mwadams) [reply Solutions/Menes.Abstractions/Menes/Exceptions/OutputBuilderNotFoundException.cs]: Fixed typos in the update.

**Pull request merged**